src/app/settings/admin/presets/dialog/steps/settings/provider/vmware-cloud-director/component.ts

```diff
@@ -0,0 +1,88 @@
+// Copyright 2020 The Kubermatic Kubernetes Platform contributors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Component, forwardRef, OnDestroy, OnInit} from '@angular/core';
+import {FormBuilder, NG_VALIDATORS, NG_VALUE_ACCESSOR, Validators} from '@angular/forms';
+import {PresetDialogService} from '@app/settings/admin/presets/dialog/steps/service';
+import {VMwareCloudDirectorPresetSpec} from '@shared/entity/preset';
+import {BaseFormValidator} from '@shared/validators/base-form.validator';
+import {merge, of} from 'rxjs';
+import {distinctUntilChanged, takeUntil} from 'rxjs/operators';
+
+export enum Controls {
+  Username = 'username',
+  Password = 'password',
+  Organization = 'organization',
+  Vdc = 'vdc',
+  OvdcNetwork = 'ovdcNetwork',
+}
+
+@Component({
+  selector: 'km-vmware-cloud-director-settings',
+  templateUrl: './template.html',
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => VMwareCloudDirectorSettingsComponent),
+      multi: true,
+    },
+    {
+      provide: NG_VALIDATORS,
+      useExisting: forwardRef(() => VMwareCloudDirectorSettingsComponent),
+      multi: true,
+    },
+  ],
+})
+export class VMwareCloudDirectorSettingsComponent extends BaseFormValidator implements OnInit, OnDestroy {
+  readonly Controls = Controls;
+
+  constructor(private readonly _builder: FormBuilder, private readonly _presetDialogService: PresetDialogService) {
+    super();
+  }
+
+  ngOnInit(): void {
+    this.form = this._builder.group({
+      [Controls.Username]: this._builder.control('', Validators.required),
+      [Controls.Password]: this._builder.control('', Validators.required),
+      [Controls.Organization]: this._builder.control('', Validators.required),
+      [Controls.Vdc]: this._builder.control('', Validators.required),
+      [Controls.OvdcNetwork]: this._builder.control('', Validators.required),
+    });
+
+    this.form.valueChanges
```

Comment:
can we move to `_initSubscriptions` method?

Reply:
Keeping it consistent with other provider components